File: src/altgraph_web_app/Models/Imdb/GraphStatsStruct.cs
namespace altgraph_web_app.Models
{
  public class GraphStatsStruct
  {
    public string Date { get; set; } = DateTime.Now.ToString();
    public int VertexCount { get; set; }
    public int EdgeCount { get; set; }
    public long ElapsedMs { get; set; }
    public long RefreshMs { get; set; }
    public DateTime? RefreshDate { get; set; } = null;
    public string RefreshSource { get; set; } = string.Empty;
    public double Epoch { get; set; }
    public double TotalMb { get; set; }
    public double FreeMb { get; set; }
    public double MaxMb { get; set; }
    public double PctFree { get; set; }
  }
}

File: java/altgraph_web_app/src/main/java/org/cjoakim/cosmos/altgraph/web/controller/struct/GraphStatsStruct.java
package org.cjoakim.cosmos.altgraph.web.controller.struct;

import lombok.Data;

import java.util.Date;

/**
 * Instances of this class are serialized as JSON to pass these values to
 * the JavaScript UI.
 * Chris Joakim, Microsoft, November 2022
 */

@Data
public class GraphStatsStruct {

    private String date = new Date().toString();

    // JGraphT graph info
    private int    vertexCount;
    private int    edgeCount;
    private long   elapsedMs;
    private long   refreshMs;
    private Date   refreshDate;
    private String refreshSource;

    // JVM Memory info
    private double epoch;
    private double totalMb;
    private double freeMb;
    private double maxMb;
    private double pctFree;
}
